addon/-private/factory-for.js

```diff
@@ -2,13 +2,30 @@
   Utility to use `ApplicationInstance#factoryFor` if available (2.12+) and
   fallback to private `_lookupFactory` otherwise
 */
-export default function factoryFor(owner, type) {
-  let factory;
+class LegacyFactoryForWrapper {
```

Comment:
Mocks the same public API that we rely on so we can consistently use class/create throughout the rest of the app.